### PR TITLE
BAK-1246 - add script for importing postfixes

### PR DESCRIPTION
**Pull request by diversemix**
### Script output
```
Importing document postfixes...
[{'id': 3057, 'postfix': 'Indonesian'}, {'id': 3058, 'postfix': 'English'}, {'id': 3059, 'postfix': ''}, {'id': 3063, 'postfix': ''}, {'id': 3064, 'postfix': ''}, {'id': 3065, 'postfix': ''}, {'id': 3066, 'postfix': 'English'}, {'id': 3067, 'postfix': 'Russian'}, {'id': 3068, 'postfix': 'Russian'}, {'id': 3069, 'postfix': 'English'}, {'id': 3070, 'postfix': ''}, {'id': 3071, 'postfix': 'Bulgarian'}, {'id': 3072, 'postfix': 'English'}, {'id': 3073, 'postfix': 'Bulgarian'}, {'id': 3074, 'postfix': 'English'}, {'id': 3075, 'postfix': 'Bulgarian'}, {'id': 3076, 'postfix': 'English'}, {'id': 3077, 'postfix': 'PDF'}, {'id': 3078, 'postfix': 'HTML'}]

```
### Test verification
```
$ psql -c "select postfix from document where id in (3057,3058,3078)"
  postfix   
------------
 Indonesian
 English
 HTML
(3 rows)

```

### CODE REVIEW

File: backend/scripts/import_doc_postfix/import_doc_postfix.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    if len(sys.argv) != 2:
+        print("Require a CSV file to import")
+        sys.exit(1)
+
+    csv_reader = validated_input(open(sys.argv[1]))
+    mappings = [
+        {
+            "import_id": import_id_from_csv_row(row),
+            "id": db.query(Document.id)
+            .filter(Document.import_id == import_id_from_csv_row(row))
```
Comment (diversemix) [inline backend/scripts/import_doc_postfix/import_doc_postfix.py:26]: I tried to do this in one up-front query but reducing the array got messy - and this way works :smile: 

Happy to hear any suggestions you might have on improving this though :+1:

Comment (joel-wright) [reply backend/scripts/import_doc_postfix/import_doc_postfix.py:26]: Hmmmm, I imagined we'd get all the document objects in one query, sort the list by the id part of the import id, then run through the list looking for duplicates...

Comment (joel-wright) [reply backend/scripts/import_doc_postfix/import_doc_postfix.py:26]: Sorry about my last comment, I thought I was looking at the relationships script... I should try to sleep! I'm still a little confused about how this works though... why are we building an ID to postfix mapping, when we can create an `import_id` to `postfix` mapping directly from the CSV, and we have a database index on `import_id`. We really never need to touch database level ids at all do we?

Comment (diversemix) [reply backend/scripts/import_doc_postfix/import_doc_postfix.py:26]: Yeah we could do it that way... I just liked how you could reduce it down to one single "bulk" update call, if you reference your updated by PK.

Comment (diversemix) [reply backend/scripts/import_doc_postfix/import_doc_postfix.py:26]: so we go...
- csv -> import_id
- import_id -> id

The last step so we can make one call to update everything with the postfix info

File: backend/app/core/validation/cclw/law_policy/process_csv.py [hunk 2]
```diff
@@ -214,6 +214,10 @@ def validated_input(csv_file: TextIOWrapper) -> DictReader:
 _ValidMetadata = Mapping[str, Mapping[str, Collection[str]]]
 
 
+def import_id_from_csv_row(row: Mapping[str, str]) -> str:
+    return f"CCLW.{row[CATEGORY_FIELD].strip()}.{row[ACTION_ID_FIELD].strip()}.{row[DOCUMENT_ID_FIELD].strip()}"
```
Comment (joel-wright) [inline backend/app/core/validation/cclw/law_policy/process_csv.py:218]: Is the category bit here safe? there's a bit more going on when calculating `presented_category`

Comment (joel-wright) [reply backend/app/core/validation/cclw/law_policy/process_csv.py:218]: which now I look no longer seems to be used...

**Review by joel-wright — APPROVED** [review]
Looks good, just that one question around the logic for `presented_category` that no longer seems to be used when calculating import ids

File: backend/app/core/validation/cclw/law_policy/process_csv.py (hunk 2 above)
Comment (diversemix) [reply backend/app/core/validation/cclw/law_policy/process_csv.py:218]: Is this an issue?

Comment (joel-wright) [reply backend/app/core/validation/cclw/law_policy/process_csv.py:218]: it's probably fine

**Review by joel-wright — APPROVED** [review]

**Pull request merged**